code/modules/clothing/glasses/glasses.dm

```diff
@@ -244,35 +245,47 @@
 
 /obj/item/clothing/glasses/sunglasses/noir/proc/toggle_noir()
 	var/list/difference = difflist(usr.client.color, color_view)
+	var/mob/living/carbon/human/user = null
 
 	if(!noir_mode)
-		if(color_view && usr.client && (!usr.client.color || difference))
-			animate(usr.client, color = color_view, time = 10)
-			noir_mode = 1
+		if(ishuman(usr))

```

Comment:
A sanity check like this belongs *outside* the two codepaths it's currently copy-pasted into. As does the `user = usr` assignment.